apps/bc-wallet-api-server/src/utils/auth.ts

```diff
@@ -1,32 +1,55 @@
 import { Buffer } from 'buffer'
 import fetch from 'cross-fetch'
+import process from 'node:process'
+import { Action, UnauthorizedError } from 'routing-controllers'
 
 export function checkRoles(token: Token, roles: string[]) {
   if (token && !roles.length) return true
   return !!(token && roles.find((role) => token.hasRole(role)))
 }
 
+export async function authorizationChecker(action: Action, roles: string[]): Promise<boolean> {
+  const authHeader: string = action.request.headers['authorization']
+  if (!authHeader || !authHeader.startsWith('Bearer ')) {
+    throw new UnauthorizedError('Missing or malformed Authorization header')
+  }
+
+  try {
+    const accessToken = authHeader.split(' ')[1]
+    // Introspect the access token
+    if (!(await isAccessTokenValid(accessToken))) {
+      return false
+    }
+    const token = new Token(accessToken, `${process.env.CLIENT_ID}`)
+    // Realm roles must be prefixed with 'realm:', client roles must be prefixed with the value of clientId + : and
+    // User roles which at the moment we are not using, do not need any prefix.
+    return checkRoles(token, roles)
+  } catch (e) {
+    throw new UnauthorizedError(e.message)
+  }
+}
+
 export async function isAccessTokenValid(token: string): Promise<boolean> {
-  const authorization = 'Basic ' + Buffer.from(`${process.env.CLIENT_ID}:${process.env.CLIENT_SECRET}`).toString('base64')
-  return fetch(
-        `${process.env.AUTH_SERVER_URL}/realms/${process.env.REALM}/protocol/openid-connect/token/introspect`,
-        {
-          method: 'POST',
-          headers: {
-            'Content-Type': 'application/x-www-form-urlencoded',
-            Authorization: authorization,
-          },
-          body: new URLSearchParams({
-            token: token,
-            client_id: `${process.env.CLIENT_ID}`,
-            client_secret: `${process.env.CLIENT_SECRET}`,
-          }),
-        },
-      ).then(checkResponse).then(response => response.json().then(data => data.active))
+  const authorization =
+    'Basic ' + Buffer.from(`${process.env.CLIENT_ID}:${process.env.CLIENT_SECRET}`).toString('base64')
+  return fetch(`${process.env.AUTH_SERVER_URL}/realms/${process.env.REALM}/protocol/openid-connect/token/introspect`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+      Authorization: authorization,
+    },
+    body: new URLSearchParams({
+      token: token,
+      client_id: `${process.env.CLIENT_ID}`,
```

Comment:
Please sync up with other env vars used by other packages in the repos we do not have to define them multiple times in the deployment.

> OIDC_CLIENT_ID=your_keycloak_id
OIDC_CLIENT_SECRET=your_keycloak_secret
OIDC_ISSUER_URL=https://your-auth-server.com/auth/realms/YourRealm
OIDC_REALM=YourRealm
OIDC_TRUST_HOST=true